### PR TITLE
Vld

### PR DESCRIPTION
**Pull request by olehmv**

### CODE REVIEW

**Comment by otataryn** [comment]
Empty PR description. 
PR name is indescriptive. 
Commits name are not really descriptive either.

**Comment by olehmv** [comment]
> Empty PR description.
> PR name is indescriptive.
> Commits name are not really descriptive either.
Ok, thanks.

File: src/main/scala/rtr/Retry.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        case x :: xs => {
+          Thread.sleep(x.toMillis)
+          retry(block, acceptResult, xs)
+        }
+      }
+    }
+  }
+
+  def retryf[A](block: () => Future[A], acceptResult: A => Boolean, retries: List[FiniteDuration]): Future[A] = {
+   block().map(a=>retry(()=>a,acceptResult,retries))
```
Comment (azhur) [inline src/main/scala/rtr/Retry.scala]: 1. retryF should call itself and be recursive.
2. your version will evaluate block() result **only once** and pass it to retry. Imagine you pass to retryf a httpcall function: in this case you will make only one http request which is not what really we need, we need to evaluate the function at each retry step.

File: src/main/scala/vld/Validator.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+object Validator {
+
+  implicit class PersonValidator(person: Person) {
+    def validate(v: Validator[Person]): Either[String, Person] = {
+      v.validate(person)
+    }
+  }
+
+  implicit class DefaultPersonValidator(person: Person) {
```
Comment (azhur) [inline src/main/scala/vld/Validator.scala:67]: there is no need to do DefaultXXX, and also no need to do it for every class. All you need to do is:
```scala
implicit class AnyValidator[T](t: T) {
    def validate(implicit v: Validator[T]): Either[String, T] = {
      v.validate(t)
    }
  }
```
then you either provide implicit validator in scope so you can just do `"".validate` or pass explicit validator `"".validate(nonEmpty)`

Comment (olehmv) [reply src/main/scala/vld/Validator.scala:67]: Ok,thanks.

**Pull request merged**